Homework/Task2_3/Training.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2_3
+{
+    public class Training : ICloneable
+    {
+        private List<TrainingContent> contents = new List<TrainingContent>();
+
+        public void Add(TrainingContent content)
+        {
+            contents.Add(content);
+        }
+
+        public bool IsPractical()
+        {
+            foreach (var content in contents)
+            {
+                if (content is not PracticalLesson)
+                    return false;
+            }
+            return contents.Count > 0;
+        }
+
+        public object Clone()
+        {
+            Training clonedTraining = new Training();
+            foreach (var content in contents)
+            {
```

Comment:
implement  TrainingContent Clone() {}  for Lecture and PracticalLesson, and simplify
clonedTraining.Add(content.Clone())

Reply:
Implemented